kyaml/yaml/rnode.go

```diff
@@ -248,6 +443,391 @@ func (rn *RNode) IsNilOrEmpty() bool {
 		IsYNodeZero(rn.YNode())
 }
 
+// AsOrDie converts a RNode to the desired typed object. ptr must
+// be a pointer to a typed object. It will panic if it encounters an error.
+func (rn *RNode) AsOrDie(ptr interface{}) {
+	if err := rn.As(ptr); err != nil {
+		panic(err)
+	}
+}
+
+// As converts a RNode to the desired typed object. ptr must be
+// a pointer to a typed object.
+func (rn *RNode) As(ptr interface{}) error {
+	if rn.IsNil() {
+		return ErrRNodeNotFound{}
+	}
+	if ptr == nil || reflect.ValueOf(ptr).Kind() != reflect.Ptr {
+		return fmt.Errorf("ptr must be a pointer to an object")
+	}
+
+	// The built-in types only have json tags. We can't simply do mv.Node().Decode(ptr),
+	// since it use the lowercased field name by default if no yaml tag is specified.
+	// This affects both k8s built-in types (e.g. appsv1.Deployment) and any types
+	// that depends on built-in types (e.g. metav1.ObjectMeta, corev1.PodTemplate).
+	// To work around it, we rely on the json tags. We first convert mv to json
+	// and then unmarshal it to ptr.
+	j, err := rn.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(j, ptr)
+	return err
+}
+
+// Get gets the value for a nested field located by fields. A pointer must be
+// passed in, and the value will be stored in ptr. ptr can be any concrete type
+// (e.g. string, []corev1.Container, []string, corev1.Pod, map[string]string) or
+// a RNode. RNode should be used if you are dealing with comments that is more
+// than what LineComment, HeadComment, SetLineComment and SetHeadComment can
+// handle. It returns if the field is found and a potential error.
+func (rn *RNode) Get(ptr interface{}, fields ...string) (bool, error) {
```

Comment:
Some questions about this one:

1. Do we need to return both `found` and an error? Here we are giving users two values that they need to check after calling this method, but it seems like it would be enough to just have an error here. 

Is there any scenario where there is no value found and there is also no error? If yes, IMO the value not being found should throw an error.  In the scenario that the value was found but there is an error, I think it would be enough to throw an error such as `found value %s, but received error %w`. WDYT?  

2. What is the benefit of having the user pass in an interface rather than returning the found value?

Reply:
Re. 1, the current behavior is that if no value is found but everything else is fine, it returns `false, nil`. An alternative is that to return only one error. If not found, return a typed error e.g. ErrFieldNotFound.

Re. 2, it can allow users to pass in the desired type e.g. https://github.com/kubernetes-sigs/kustomize/pull/4462/files#diff-c8257adc9e5e8838a4b1d0e35ea30fbb64af63849976fab8e023aa4c6570b7d3R949.
It's better than return an `interface{}` type and let the user do the type assertion.